Sources/ElevenLabs/Internal/Conversation/Startup/ConversationStartupOrchestrator.swift

```diff
@@ -0,0 +1,236 @@
+        options: ConversationOptions,
+        provider: any ConversationDependencyProvider,
+        onStateChange: @escaping (ConversationStartupState) -> Void,
+        onRoomConnected: @escaping (Room) -> Void
+    ) async throws -> StartupResult {
+        let startTime = Date()
+        var metrics = ConversationStartupMetrics()
+
+        // Cache connection manager for synchronous access throughout startup
+        let connectionManager = await provider.connectionManager()
+
+        let agentId = extractAgentId(from: auth)
+        let context = ["agentId": agentId]
+        logger.info("Starting conversation startup sequence", context: context)
+
+        // Step 1: Parallel Token Resolution & Mic Permission
+        onStateChange(.resolvingToken)
+
+        var connectionDetails: TokenService.ConnectionDetails?
+        let tokenService = await provider.tokenService
+        let tokenStep = TokenResolutionStep(
+            tokenService: tokenService,
+            auth: auth,
+            logger: logger
+        ) { details in
+            connectionDetails = details
+        }
+
+        // 1. Start fetching token (Main Task)
+        async let tokenResult: Void = tokenStep.execute()
+
+        // 2. Start requesting permission (Background Task)
+        // This allows the system permission prompt to appear immediately
+        async let permissionResult: Bool = {
+            guard !options.conversationOverrides.textOnly else { return true }
+            return await withCheckedContinuation { continuation in
+                #if os(iOS)
+                AVAudioSession.sharedInstance().requestRecordPermission { granted in
+                    continuation.resume(returning: granted)
+                }
+                #elseif os(macOS)
+                switch AVCaptureDevice.authorizationStatus(for: .audio) {
+                case .authorized:
+                    continuation.resume(returning: true)
+                case .notDetermined:
+                    AVCaptureDevice.requestAccess(for: .audio) { granted in
+                        continuation.resume(returning: granted)
+                    }
+                case .denied, .restricted:
+                    continuation.resume(returning: false)
+                @unknown default:
+                    continuation.resume(returning: false)
+                }
+                #endif
+            }
+        }()
+
+        let tokenStart = Date()
+        let permissionGranted: Bool
+        do {
+            // Wait for both to complete
+            try await tokenResult
+            permissionGranted = await permissionResult
+
+            metrics.tokenFetch = Date().timeIntervalSince(tokenStart)
+        } catch {
+            metrics.tokenFetch = Date().timeIntervalSince(tokenStart)
+            metrics.total = Date().timeIntervalSince(startTime)
+            throw StartupFailure.token(error as? ConversationError ?? .connectionFailed(error), metrics)
```

Comment:
We might want to handle Cancellation errors separately for all these longer running tasks, as they will get caught and converted to `StartupFailure`